.github/workflows/ci.yml

```diff
@@ -25,3 +25,10 @@ jobs:
           go mod tidy
           go build
           go test -v ./...
+          
+      - name: golangci-lint
```

Comment:
I would suggest that we run the linter checks before the building and running tests. The reason is, linter checks are relatively fast compared to build and test. So we will consume less CI minutes if we just fail fast with linter checks before running build and test.

Reply:
Also can you verify if the `golangci/golangci-lint-action@v3` lints only the PR diffs or entire code base? It will be great if we can lint only the changed files to start with. This is because we will probably have quite a lot of linter failures in existing code (tech debt) which we need to fix incrementally.